src/wallet/sqlite.cpp

```diff
@@ -607,7 +612,7 @@ std::unique_ptr<DatabaseCursor> SQLiteBatch::GetNewPrefixCursor(Span<const std::
 bool SQLiteBatch::TxnBegin()
 {
     if (!m_database.m_db || sqlite3_get_autocommit(m_database.m_db) == 0) return false;
-    int res = sqlite3_exec(m_database.m_db, "BEGIN TRANSACTION", nullptr, nullptr, nullptr);
+    int res = m_exec_handler && m_exec_handler->Exec(m_database, "BEGIN TRANSACTION");
```

Comment:
In commit "sqlite: add ability to interrupt statements" (177ff954c1221a00d4bf5ea63e6bfc415ba706c8)

This doesn't seem right because the `&&` operator will convert the integer return value into a bool. Also if m_exec_handler is null the `res` value will be 0 which is SQLITE_OK so the call will appear to succed.

It probably would make more sense to do `int res = Assert(m_exec_handler->Exec`

Reply:
yes, fixed. Thanks. God knows why I did that.. it seems that I wasn't focused.